runner/operator/roslin_operator/bin/test_pair_request.py

```diff
@@ -0,0 +1,211 @@
+import os
```

Comment:
Also lets create a `runner/tests/` package and put all tests for runner there. It is batter practice to keep all tests at the same place.

And thanks for the tests :)

Reply:
if you put them all in `runner/tests` will it make it more difficult to track which tests go with which submodules? Especially since we have a heavily nested dir structure here with submodules and functions with the same name across the operators. I usually try to keep tests adjacent to their corresponding `.py` files so its not ambiguous which test file goes with which `.py` file, also makes it a little less complicated to import the module for testing if its adjacent. I think `python manage.py test` automatically discovers all tests?

Reply:
You are right `python manage.py test` do discover all the tests. I mean I would prefer following the Django best practice for this, which is keeping all the tests together. In cases you mentioned when you have a nested directory structure, usually what people do is basically mirroring the project structure. In this case I probably wouldn't go all the way to `runner/tests/operator/roslin_operator/bin/test_pair_request.py` but `runner/tests/operator/roslin_operator/test_roslin_operator.py`, where we could put all the testes related to roslin operator. This also helps separating tests by type so you can run only certain tests, which is harder if you have test files in all modules. Something like [this](https://dezoito.github.io/2015/09/21/how-to-test-django-applications_pt1.html)